### PR TITLE
[FEATURE] ID/PK flag `return_unexpected_index_list` that allows index value output to be suppressed

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What is this PR? 
* It's a small follow-up to the ID/PK work that introduces a `return_unexpected_index_list` that allows `unexpected_indices` output to be suppressed selectively.
* Tests for Pandas, Spark and SQL. 
* Update to documentation. 

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: docs/reference/expectations/result_format.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ The `result_format` parameter may be either a string or a dictionary which speci
     * `return_unexpected_index_query`: When running validations, a query (or a set of indices) will be returned that will
       allow you to retrieve the full set of unexpected results including any columns identified in `unexpected_index_column_names`.  Setting this value to `False` will 
       suppress the output (default is `True`).
+    * `return_unexpected_index_list`:  Setting this value to `False` will suppress `unexpected_index_list` output (default is `True`).
```
Comment (Shinnnyshinshin) [inline docs/reference/expectations/result_format.md:14]: Update to documentation

Comment (talagluck) [reply docs/reference/expectations/result_format.md:14]: Could you provide an example of when a user would want to do this? Is it for cases where a user wants the query but not the list?

**Review by anthonyburdi — COMMENTED** [review]
Looks good to me pending resolution of Tal's comment. Thank you for the helpful test docstrings.

File: docs/reference/expectations/result_format.md (hunk 1 above)
Comment (Shinnnyshinshin) [reply docs/reference/expectations/result_format.md:14]: @talagluck this was a request from a user. I believe they just wanted to use the unexpected results directly, without going through the DataDocs report

Comment (Shinnnyshinshin) [reply docs/reference/expectations/result_format.md:14]: I sent a link to the thread to Slack

**Comment by Shinnnyshinshin** [comment]
closing this PR as #7150 will output query by default. May re-open in the future if it becomes a feature request from the community.

**Pull request closed without merging**